src/main/java/com/enderio/base/common/util/ExperienceUtil.java

```diff
@@ -16,14 +16,27 @@ public class ExperienceUtil {
      * @param stopLevel
      * @return
      */
-    public static Vector2i getLevelForExpWithLeftover(int experience, int startLevel, int stopLevel) {
-        int expNeeded = getXpNeededForNextLevel(startLevel);
-        if (expNeeded < experience && startLevel < stopLevel) {
-            experience -= expNeeded;
-            startLevel +=1;
-            return getLevelForExpWithLeftover(experience, startLevel, stopLevel);
+    public static Vector2i getLevelForExpWithLeftover(long experience, int startLevel, int stopLevel) {
```

Comment:
depending on needs it might be good idea to return here something that can handle bigger leftover values, but I don't think its an issue for now